### DISCUSSION TITLE
SAVE_CONFIG

### DISCUSSION THREAD

**Discussion by Jpow1987** [dz0ny/klipper-prusa-mk3s#24] [General]
After running PID_CALIBRATE HEATER=extruder Target=170 and trying to run SAVE_CONFIG I get the following error

SAVE_CONFIG section 'extruder' option 'control' conflicts with included value

**Comment by confuzled-dev** [comment]
You are trying to save pid values that are already saved in an include elsewhere.

It is telling you that in the corresponding [include "filenamehere".cfg] there is a section called [extruder] that has an option listed as "control" likely "control: pid"

I will post a solution to this tomorrow. Also, you will want to do your pid tune at your normal print temp with your typical fan speed set. Example: if you mostly print PLA, then pid tune to your print temp and set the fan to 100% prior to running the test. If you plan to use a hotend sock, ensure it is installed prior to starting the test.

Take care

**Reply by charminULTRA** [reply]
This doesn't make sense. Why would a fresh install already have "control" listed? It's commented out of the template, and the instructions say not to include it yet until you do a PID tuning.

**Comment by confuzled-dev** [comment]
[extruder]
#To tune Pressure Advance see https://www.klipper3d.org/Pressure_Advance.html
#default is already set based on hotend, but you can further improve prints by calibrating it to your nozzle and material
#pressure_advance: 0.05
nozzle_diameter: 0.4 #Remember to change this if you change nozzle diameter.
max_extrude_only_distance: 201
min_extrude_temp: 150
#control: pid
#pid_Kp: 27.517
#pid_Ki: 1.310
#pid_Kd: 144.467

I added the items above to my p[rinter.cfg and then manually added the values after the last line where it says not to add things. This is not advised and should not be a primary method to resolve the issue

If any of the format in the bottom section is wrong, it will ignore the entire section so ensure that you respect spacing and format. TBH, you shouldn't be adding things here manually, but it worked for me.

Part of the issue is that things that are called out in the cfgs that _aren't_ printer.cfg are read only and can't be edited. I think this causes some complications. My solution was to copy the offended cfgs and create {custom_insertcfgnamehere}.cfg files and call those in the includes instead of the versions from the git. 
I currently have my repo marked as dirty and can't tell the source, so tread lightly. _there be dragons_

**Comment by confuzled-dev** [comment]
> After running PID_CALIBRATE HEATER=extruder Target=170 and trying to run SAVE_CONFIG I get the following error
> 
> SAVE_CONFIG section 'extruder' option 'control' conflicts with included value

Have you been able to correct your issue with saving PID?

**Comment by olituks** [comment]
Hello, same issue for me.
It's a very fresh installation from Klipper so I not applying any custom's.
I started, as recommended in the printer.cfg file, by the nozzle PID calibration and I get the below message.
**SAVE_CONFIG section 'extruder' option 'control' conflicts with included value**
[printer.txt](https://github.com/dz0ny/klipper-prusa-mk3s/files/13830120/printer.txt)
(I  added my printer.cfg in a txt version)

**Reply by olituks** [reply]
Same issue with the bed PID
SAVE_CONFIG section 'heater_bed' option 'control' conflicts with included value

**Reply by olituks** [reply]
To solve the SAVE_CONFIG recording for both PID I simply added in the config.cfg file the bellow lines with the PID parameter returned by my printer.

[extruder]
pid_Kp: 23.862
pid_Ki: 1.020
pid_Kd: 139.595

[heater_bed]
pid_Kp=32.683 
pid_Ki=2.758 
pid_Kd=96.823

This is the correct way to proceed?

**Comment by confuzled-dev** [comment]
This will give previously established values but I still run into saving
issues.

What I did was create all new cfg files to get around some of the saving
and read only issues.

This repo is a good starting point but I would recommend learning to bend
it to your will

**Comment by dz0ny** [comment]
After you do the calibration and before save, you will have the values printed in the console. You need to save those to printer.cfg on your own, and that will override those in stock config.

I've updated the defaults to make it clearer.

**Comment by dz0ny** [comment]
Klipper supports multiple definitions of same config (to extend an override), but not when using SAVE_CONMFIG command. Master was updated and from now on everyone should be good.

**Comment by confuzled-dev** [comment]
Charmin,

Tbh, this repo is not very active.

I have a working mk3 running klipper that is originally based on this repo.
If you wanted to work together to do some better documentation and
establish a better baseline, id be interested

**Reply by charminULTRA** [reply]
Cool, I'll try to get some PRs submitted when I get the chance this week.

**Comment by confuzled-dev** [comment]
If you are in the klipper discord, hit me up.

@confuzled

**Reply by confuzled** [reply]
???

**Comment by confuzled-dev** [comment]
Sorry mate. Your name is my discord name and I am on Mobile. Didn't realize
it @ you

**Reply by confuzled** [reply]
No worries
